### ISSUE TITLE
Support for postestimation -margins-

### ISSUE THREAD

**Issue by sergiocorreia** [sergiocorreia/reghdfe#32]
Currently `margins` "works" but gives wrong results. We can either set e(marginsok) to missing/empty, or add margins support. No clue how though.

``` stata
clear all
cls
sysuse auto

reghdfe price weight i.foreign, a(turn)
margins foreign

areg price weight i.foreign, a(turn)
margins foreign
```

**Comment by rdisalv2** [comment]
Hi Sergio, I was wondering if you could provide some clarification on when -margins- works and does not work in -reghdfe-. 

Continuous interactions appear to work just fine -- things only appear to be wrong when there are factor variables. My use-case is pretty much always getting average marginal effects after some c.x#c.y so I'd like to be able to run -margins- after reghdfe for this. (Of course these AMEs can be obtained manually pretty easily though, so I suppose I could just go that route. -margins- is addicting though.)

```
// these appear to work (continuous interactions only):
clear all
cls
sysuse auto

areg price c.weight##c.headroom, a(turn)
margins, dydx(weight) at(headroom=(1.5(0.5)5))

reghdfe price c.weight##c.headroom, a(turn)
margins, dydx(weight) at(headroom=(1.5(0.5)5))

areg price weight headroom c.weight#c.headroom, a(turn)
margins, dydx(weight) at(headroom=(1.5(0.5)5))

reghdfe price weight headroom c.weight#c.headroom, a(turn)
margins, dydx(weight) at(headroom=(1.5(0.5)5))

// these don't (factor interactions): 

clear all
cls
sysuse auto

areg price c.weight##i.foreign, a(turn)
margins, dydx(weight) at(foreign = 1)

reghdfe price c.weight##i.foreign, a(turn)
margins, dydx(weight) at(foreign = 1)
```

**Comment by sergiocorreia** [comment]
I agree that it's something worth to have, and probably is not that simple to implement, but the backend of `margins` is not well documented, so I'm not entirely sure of what do I need to be able to implement it.

For instance, I tried reading the manual for `margins`, as well as the source code (`viewsource margins.ado`), and random google searches such as [this one](https://stackoverflow.com/questions/28470632/how-to-make-stata-margins-work-for-user-written-model).

It seems that there are some special ados that you might want to create, and then store in `e(margins_prolog)`, `e(marginsok)`, `e(marginsnotok)`, etc. but I'm not really sure. The best I've found so far is a brief announcement [here](https://www.stata.com/stata11/prog.html). If you find something, or maybe someone from Statacorp suggests something, then for sure I'm up for implementing it!
